File: strings/remove-palindromic-subsequences/remove-palindromic-subsequences.go
package removepalindromicsubsequences

func removePalindromeSub(s string) int {
	if len(s) == 0 {
		return 0
	}

	left := 0
	right := len(s) - 1

	for left < right {
		if s[left] != s[right] {
			return 2
		}
		left++
		right--
	}

	return 1

}

File: strings/remove-palindromic-subsequences/remove-palindromic-subsequences.js
/**
 * @param {string} s
 * @return {number}
 */
function removePalindromeSub(s) {
  if (!s) {
    return 0;
  }

  let left = 0;
  let right = s.length - 1;
  while (left < right) {
    if (s[left] !== s[right]) {
      return 2;
    }

    left++;
    right--;
  }
  return 1;
}
